### PR TITLE
Makedata and Checkdata for analyzers tab

### PR DESCRIPTION
**Pull request by MDAlFattah**

### CODE REVIEW

File: test_data/makedata_suites/600_analyzers.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      db._query(`RETURN TOKENS("foobar", "trigram")`).toArray();
+      
+      print(`Checking trigram analyzer properties.`)
+      a.analyzer("_system::trigram").properties();
+    },
+
+    clearData: function (options, isCluster, isEnterprise, dbCount, loopCount, readOnly) {
+      print(`checking data ${dbCount} ${loopCount}`);
+      try {
+        a.remove("trigram");
```
Comment (dothebart) [inline test_data/makedata_suites/600_analyzers.js:49]: fix the name pls.

File: test_data/makedata_suites/600_analyzers.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    checkData: function (options, isCluster, isEnterprise, dbCount, loopCount, readOnly) {
+      print(`checking data ${dbCount} ${loopCount}`);
+      // Check analyzer:  analyzers.analyzer("trigram").properties();
+      let analyzer1 = `ngram_${loopCount}`;
+      print(`Listing all analyzers in current database`)
+      a.toArray();
+
+      print(`Checking number of analyzer is correct`)
+      if (a.toArray().length !== 14) {
```
Comment (dothebart) [inline test_data/makedata_suites/600_analyzers.js:34]: search the array for an analyzer of the matching name. 
example code:
```
    return global.theInstanceManager.arangods.filter((instance) => instance.instanceRole === role);
```
(unrelated, just found in cut'n'paste buffer.

**Comment by dothebart** [comment]
this is a good start!
now we need 
- 1 file with all analyzers available since 3.8
- 1 file (the already added one) filled with all analyzers that became possible with 3.9
- 1 file for devel

File: test_data/makedata_suites/600_analyzers.js (hunk 1 above)
Comment (MDAlFattah) [reply test_data/makedata_suites/600_analyzers.js]: done.

**Comment by dothebart** [comment]
please add more error handling, as stated. 
please also lint using the arangodb source `./utils/jslint.sh [filename]`; from a first glance tons of `;` are missing - Horay for python ;-)

File: test_data/makedata_suites/607_analyzers.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          stopwords: []
+        }, ["frequency", "norm", "position"]);
+        }, function () {
+          return a.analyzer(`text_0`);
+        });
+    },
+    checkDataDB: function (options, isCluster, isEnterprise, dbCount, readOnly) {
+      print(`checking data ${dbCount}`);
+      print(`Listing all analyzers in current database`);
+      a.toArray();
```
Comment (dothebart) [inline test_data/makedata_suites/607_analyzers.js:33]: wrap with print

File: test_data/makedata_suites/607_analyzers.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      print(`Checking number of analyzer is correct`);
+      if (a.toArray().length !== 14) {
```
Comment (dothebart) [inline test_data/makedata_suites/607_analyzers.js:35]: filter for the exact analyzer you're searching for

File: test_data/makedata_suites/607_analyzers.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        throw new Error("Analyzer not created!");
+      }
+      progress();
+
+      if (a.analyzer("text_0") == null) {
```
Comment (dothebart) [inline test_data/makedata_suites/607_analyzers.js:40]: create string respecting count parameter

File: test_data/makedata_suites/607_analyzers.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      try {
+        const array = a.toArray();
+        for (let i = 0; i < array.length; i++) {
+          const name = array[i];
+          if (name == `text_0`) {
+            a.remove(`text_0`);
+          }
+        }
+        // checking created analyzer is deleted or not
+        if (a.analyzer("text_0") != null) {
```
Comment (dothebart) [inline test_data/makedata_suites/607_analyzers.js:74]: generate name

File: test_data/makedata_suites/608_analyzers.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+              min: 2, max: 2, preserveOriginal: false, streamType: "utf8" } }] }, ["frequency", "norm", "position"]);
+        }, function () {
+          return a.analyzer(`pipeline_0`);
+        });
+    },
+    checkDataDB: function (options, isCluster, isEnterprise, dbCount, readOnly) {
+      print(`checking data ${dbCount}`);
+      // Check analyzer:  analyzers.analyzer("trigram").properties();
+      print(`Listing all analyzers in current database`);
+      a.toArray();
```
Comment (dothebart) [inline test_data/makedata_suites/608_analyzers.js:30]: wrap in print

File: test_data/makedata_suites/608_analyzers.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+      print(`Checking number of analyzer is correct`);
+      if (a.toArray().length !== 14) {
+        throw new Error("Analyzer not created!");
+      }
+      progress();
+
+      if (a.analyzer("pipeline_0") == null) {
```
Comment (dothebart) [inline test_data/makedata_suites/608_analyzers.js:37]: generate name

File: test_data/makedata_suites/608_analyzers.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+          "WN",
+          "N ",
+          " F",
+          "FO",
+          "OX"
+        ]
+      ];
+
+      // print(`Create and use a pipeline Analyzer with preserveOriginal disabled:`)
+      let pipelineArray = db._query(`RETURN TOKENS("Quick brown foX", "pipeline_0")`).toArray();
```
Comment (dothebart) [inline test_data/makedata_suites/608_analyzers.js:67]: generate name

**Review by dothebart — CHANGES_REQUESTED** [review]
- make sure you generate the name of the item with the db-counter in all places
- wrap print or delete if toArray() aren't needed
- filter for the analyzer you want to inspect instead of asserting on the total count of analyzers in the db

**Comment by dothebart** [comment]
https://jenkins01.arangodb.biz/view/Upgrade/job/Upgrade-3.9-to-devel.Linux/425/ launched.

**Pull request merged**